### PR TITLE
Fix socketio connection for imjoy-core 

### PR DESCRIPTION
**Pull request by oeway**
- Make the rpc connection work with the new imjoy-core implementation here: imjoy-team/imjoy-engine#106

### CODE REVIEW

File: python/imjoy_rpc/connection/socketio_connection.py [hunk 1]
```diff
@@ -57,39 +58,46 @@ def setup():
...
-
             config = self.default_config.copy()
             cfg = self.default_config
             if cfg.get("credential_required") is not None:
-                result = config.verify_credential(cfg["credential"])
+                result = config["verify_credential"](cfg["credential"])
                 cfg["auth"] = result["auth"]
-            cfg["id"] = config["id"]
-            rpc = RPC(connection, self.rpc_context, config=cfg, codecs=self._codecs,)
+            cfg["id"] = config.get("id")
```
Comment (MartinHjelmare) [inline python/imjoy_rpc/connection/socketio_connection.py:94]: Is it ok for the id to be `None`?

File: python/setup.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def read(name):
     extras_require={
         "full": ["numpy", "zarr"],
         "jupyter": ["numpy", "ipykernel"],
-        "socketio": ["numpy", "python-socketio[client]"],
+        "socketio": ["numpy", "python-socketio[client]==5.0.4", "python-engineio==4.0.0"],
```
Comment (MartinHjelmare) [inline python/setup.py:49]: Why are we not using the latest version of python-socketio?

**Comment by MartinHjelmare** [comment]
What is the issue we're fixing with the PR?

File: python/setup.py (hunk 2 above)
Comment (oeway) [reply python/setup.py:49]: Somehow we have some issues connecting to the server here: https://github.com/imjoy-team/imjoy-engine/pull/106

**Comment by oeway** [comment]
> What is the issue we're fixing with the PR?

We want to make it work with the new imjoy-core implementation here: https://github.com/imjoy-team/imjoy-engine/pull/106

File: python/setup.py (hunk 2 above)
Comment (MartinHjelmare) [reply python/setup.py:49]: Should we let imjoy-engine decide which version to use then? It is installing this package, right? We should just add socketio as a general requirement.

Comment (MartinHjelmare) [reply python/setup.py:49]:
```suggestion
        "socketio": ["numpy", "python-socketio[client]"],
```

Comment (oeway) [reply python/setup.py:49]: The issue is the rpc library may not run on the same machine as the imjoy-engine. To make sure they work with each other, they need to have the same library versions.

File: python/imjoy_rpc/connection/socketio_connection.py (hunk 1 above)
Comment (oeway) [reply python/imjoy_rpc/connection/socketio_connection.py:97]: That's ok, we will generate it later.

File: python/setup.py (hunk 2 above)
Comment (MartinHjelmare) [reply python/setup.py]: What will run the rpc library in the other machine then?

Comment (MartinHjelmare) [reply python/setup.py]: If the rpc package version is not the same in the two different machines we don't know that the two machines have the same pinning here anyway.

Comment (oeway) [reply python/setup.py]: At least we know the server will have the same version now. The issue is, if I just do without, I cannot get the connection work. I would pin it for now and explore later to see what the exact issue.

Comment (MartinHjelmare) [reply python/setup.py]: But what is using this package in the server?

Comment (oeway) [reply python/setup.py]: the imjoy-rpc will be imported by imjoy.runner on an independent machine (other than the machine that runs imjoy.server).

Comment (oeway) [reply python/setup.py]: Would this be better?
```
"python-socketio[client]>=5.0.4",
"python-engineio>=4.0.0",
```

Comment (MartinHjelmare) [reply python/setup.py]: So then we should pin or constrain the version in imjoy.runner and imjoy.server.

Comment (MartinHjelmare) [reply python/setup.py]: Yes, a minimum version is ok here.

Comment (oeway) [reply python/setup.py]: In that case yes, but that won't cover the case when one just use imjoy_rpc directly like this:
```
from imjoy_rpc import connect_to_server, api

connect_to_server({"name": "my plugin", "server_url": "https://api.imjoy.io"})

class ImJoyPlugin:
    async def setup(self):
        pass

    async def run(self, ctx):
        await api.log("hello world.")

api.export(ImJoyPlugin(), config={"name": "test-plugin", "workspace": "123"})

```

**Review by MartinHjelmare — APPROVED** [review]

**Pull request merged**